multiapps-controller-core/src/test/java/org/cloudfoundry/multiapps/controller/core/cf/TaggingRequestFilterTest.java

```diff
@@ -0,0 +1,169 @@
+import org.springframework.web.reactive.function.client.ExchangeFunction;
+import org.springframework.web.reactive.function.client.ExchangeStrategies;
+
+import reactor.core.publisher.Mono;
+
+public class TaggingRequestFilterTest {
+
+    private static final String TEST_VERSION_VALUE = "1.58.0";
+    private static final String TEST_ORG_VALUE = "faceorg";
+    private static final String TEST_SPACE_VALUE = "myspace";
+    private HttpHeaders httpHeaders;
+    private ClientRequest clientRequest;
+    @Mock
+    private ExchangeFunction nextFilter;
+    @Mock
+    private ApplicationConfiguration configuration;
+
+    @Before
+    public void setUp() throws Exception {
+        httpHeaders = new HttpHeaders();
+        clientRequest = initializeClientRequest();
+        MockitoAnnotations.openMocks(this)
+                          .close();
+    }
+
+    private ClientRequest initializeClientRequest() {
+        return new ClientRequest() {
+
+            @Override
+            public Map<String, Object> attributes() {
+                return null;
+            }
+
+            @Override
+            public BodyInserter<?, ? super ClientHttpRequest> body() {
+                return null;
+            }
+
+            @Override
+            public MultiValueMap<String, String> cookies() {
+                return null;
+            }
+
+            @Override
+            public HttpHeaders headers() {
+                return httpHeaders;
+            }
+
+            @Override
+            public String logPrefix() {
+                return null;
+            }
+
+            @Override
+            public HttpMethod method() {
+                return null;
+            }
+
+            @Override
+            public URI url() {
+                return null;
+            }
+
+            @Override
+            public Mono<Void> writeTo(ClientHttpRequest arg0, ExchangeStrategies arg1) {
+                return null;
+            }
+        };
+    }
+
+    @Test
+    public void testInjectGenericValue() throws IOException {
+        TaggingRequestFilter testedFilter = new TaggingRequestFilter("1.58.0");
+        testedFilter.filter(clientRequest, nextFilter);
+        assertNotNull(clientRequest.headers());
+        assertTrue(clientRequest.headers()
+                                .containsKey(TaggingRequestFilter.TAG_HEADER_NAME));
+        String expectedValue = "MTA deploy-service v1.58.0";
+        Optional<String> foundValue = clientRequest.headers()
+                                                   .get(TaggingRequestFilter.TAG_HEADER_NAME)
+                                                   .stream()
+                                                   .filter(value -> value.equals(expectedValue))
+                                                   .findFirst();
+        assertTrue(foundValue.isPresent());
+    }
+
+    @Test
+    public void testInjectOrgAndSpaceValues() throws IOException {
```

Comment:
Why so many unit tests? I think that some of them tests the same thing.
You can prepare one larger unit tests that tests every tag, other option is to have 3 different unit tests for each tag